src/main/kotlin/uk/gov/justice/digital/hmpps/supportadditionalneedsapi/service/ChallengeService.kt

```diff
@@ -45,6 +47,11 @@ class ChallengeService(
   }
 
   @Transactional
+  @TimelineEvent(
```

Comment:
Ooh Sir, a little bit of AOP magic? Nice, I like it 👍 

I'm sure we considered a similar approach in LWP, but I can't remember why we didn't use it. FWIW I quite like AOP (in context of simply applying an annotation ... sometimes the impl can be ... errr .... 🤣  (I've already seen your dragon comment below 🤣 ))
AOP feels like a good fit for a cross cutting concern like this 👍